src/relay/backend/graph_executor_codegen.cc

```diff
@@ -215,24 +215,35 @@ class GraphExecutorCodegen : public backend::MemoizedExprTranslator<std::vector<
 
     if (memory_plan_.defined()) {
       // TODO(@electriclilies, @jroesch): remove UpdateMainWorkspaceSize
-      func_info =
-          relay::tec::UpdateMainWorkspaceSize(mod, targets_, memory_plan_->expr_to_storage_info);
+      tec::TargetMap tec_target_map;
+      for (const auto& pair : targets_) {
+        tec_target_map.emplace(static_cast<DLDeviceType>(pair.first->value), pair.second);
+      }
```

Comment:
Is this copy just to make sure UpdateMainWorkspaceSize doesn't have access to targets_?

Reply:
No, it's just a representation mismatch between Map<Integer,...> and unordered_map<DLDeviceType, ...>.